x/community/keeper/proposal_handler.go

```diff
@@ -15,3 +19,62 @@ func HandleCommunityPoolLendDepositProposal(ctx sdk.Context, k Keeper, p *types.
 func HandleCommunityPoolLendWithdrawProposal(ctx sdk.Context, k Keeper, p *types.CommunityPoolLendWithdrawProposal) error {
 	return k.hardKeeper.Withdraw(ctx, k.moduleAddress, p.Amount)
 }
+
+// HandleCommunityPoolProposal is a handler for executing a passed community pool proposal
+func HandleCommunityPoolProposal(ctx sdk.Context, k Keeper, p *types.CommunityPoolProposal) error {
+	logger := k.Logger(ctx)
+
+	// attempt to execute all messages within the passed proposal
+	// Messages may mutate state thus we use a cached context. If one of
+	// the handlers fails, no state mutation is written and the error
+	// message is logged.
+	cacheCtx, writeCache := ctx.CacheContext()
+	msgs, err := p.GetMsgs()
+	if err != nil {
+		return err
+	}
+
+	enabledUrls := k.GetParams(ctx).EnabledProposalMsgUrls
+	for idx, msg := range msgs {
+		if !isMsgEnabled(msg, enabledUrls) {
+			return logAndReturnProposalMsgError(logger, idx, msg, "msg not enabled via params")
+		}
+
+		// assert that the community module account is the only signer of the messages
+		signers := msg.GetSigners()
+		if len(signers) != 1 || !signers[0].Equals(k.moduleAddress) {
```

Comment:
thought we talked about making this extensible for when this proposal message is used for infrastracture pool funds but it looks like here we aren't supporting being able to initialize the keeper with a slice of whitelisted signer addresses?

Reply:
thats for the `EvmCallMsg`, which is done [here](https://github.com/Kava-Labs/kava/pull/1424/commits/022f5d7d82b616d994a127891be88d25ea4eab8d). On a proposal level, it needs to be module specific.